### PR TITLE
CHALLENGE-75: Registration users

### PR DESCRIPTION
**Pull request by vlaskiri**
The ability to register has been added, errors are displayed, a loader has been added when sending data during registration, the design has been corrected according to the layout.

### CODE REVIEW

File: src/components/registrationWindow/info/Info.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import img from '../../../images/registrationWindow/puzzles_img.png';
```
Comment (elzbthbrst) [inline src/components/registrationWindow/info/Info.jsx:2]: may be u should chose better name for this picture? i think the name of the picture should contains something about what picture showa

File: src/components/registrationWindow/registrationAndEnter/loginOrRegister/LoginAndRegistration.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import useLoginAndRegistrationLogic from './useLoginAndRegistrationLogic';
+import ModalRegistration from './ModalRegistration/ModalRegistration';
+
+export default function LoginAndRegistration({setUniKey}) {
+    const state = useLoginAndRegistrationLogic();
```
Comment (elzbthbrst) [inline src/components/registrationWindow/registrationAndEnter/loginOrRegister/LoginAndRegistration.jsx:6]: better to use destructuring like  const {
        uniKey,
        uniKeyError,
        modal,
        setModal,
        onChangeUniqueKey,
        onDisplay,
    } = useLoginAndRegistrationLogic();

File: src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import close from '../../../../../images/registrationWindow/close.svg';
+import useModalRegistrationLogic from './useModalRegistrationLogic';
+import MopsAvatars from './MopsAvatars';
+import LoadingOnFormSubmit from './LoadingOnFormSubmit';
+
+// Array avatars Mops
+const mops = [
```
Comment (elzbthbrst) [inline src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx:8]: better to create another file for this array

File: src/images/registrationWindow/close.svg
Comment (elzbthbrst) [inline src/images/registrationWindow/close.svg:1]: i think better to use react-icons library

File: src/styles/index.scss [hunk 4]
```diff
@@ -1,11 +1,11 @@
 @import "reset";
 @import url("https://fonts.googleapis.com/css2?family=Lexend:wght@100;200;300;400;500;600;700;800;900&display=swap");
 @import url("https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap");
-@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@100;200;300;400;500;600;700;800;900&display=swap");
-@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap");
+//@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@100;200;300;400;500;600;700;800;900&display=swap");
+//@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;200;300;400;500;600;700;800;900&display=swap");
```
Comment (elzbthbrst) [inline src/styles/index.scss:5]: delete this?

File: src/styles/variables.scss [hunk 5]
```diff
@@ -2,21 +2,24 @@ $color-light-white: #FCFCFC;
 $color-white: #FFF;
 $color-grey: #EBEAEF;
 $color-black: #272732;
+$color-light-black: #151515;
 $color-text-grey: #5B5B66;
-$color-border-light-gray: #aba8d4;
```
Comment (elzbthbrst) [inline src/styles/variables.scss:6]: why u decided to change this variable value?

File: src/components/registrationWindow/info/Info.jsx (hunk 1 above)
Comment (vlaskiri) [reply src/components/registrationWindow/info/Info.jsx:2]: There is only one picture. But I can rename puzzle-img.

File: src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx (hunk 3 above)
Comment (vlaskiri) [reply src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx:8]: I threw this array into a file with the useModalRegistrationLogic logic

File: src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    {src: 'https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=735&q=80', alt: 'Mops1'},
+    {src: 'https://images.unsplash.com/photo-1541364983171-a8ba01e95cfc?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80', alt: 'Mops2'},
+    {src: 'https://images.unsplash.com/photo-1517423440428-a5a00ad493e8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=749&q=80', alt: 'Mops3'},
+    {src: 'https://images.unsplash.com/photo-1517849845537-4d257902454a?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=735&q=80', alt: 'Mops4'},
+    {src: 'https://images.unsplash.com/photo-1575425186775-b8de9a427e67?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80', alt: 'Mops5'},
+]
+
+export default function ModalRegistration({onClose, setUniKey}) {
+    const state = useModalRegistrationLogic({setUniKey});
```
[the remark this answers is not in the document]
Comment (vlaskiri) [reply src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx:17]: Then there will be a lot of extra code. This is how I get the values without polluting the component

File: src/styles/index.scss (hunk 4 above)
Comment (vlaskiri) [reply src/styles/index.scss:5]: Yes

File: src/styles/registrationWindow/info.scss [hunk 7]
```diff
@@ -53,14 +56,22 @@
...
-                margin: 0 0 0 14px;
+                margin: 0 0 8px 14px;
             }
 
-            p {
-                font-size: 20px;
-                margin: 8px 0 0 14px;
+            span {
+                font-family: $font-inter;
+                font-style: normal;
```
[the remark this answers is not in the document]
Comment (vlaskiri) [reply src/styles/registrationWindow/info.scss:70]: removed, thank you)

File: src/styles/registrationWindow/modal.scss [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+                resize: none;
+            }
+        }
+
+        .nickname-error,
+        .link-error {
+            color: red;
+            padding: 5px 0;
+            margin: -30px 0 10px 0;
+            //text-shadow: 1px 1px 3px red;
```
[the remark this answers is not in the document]
Comment (vlaskiri) [reply src/styles/registrationWindow/modal.scss:82]: No, removed

File: src/styles/variables.scss (hunk 5 above)
Comment (vlaskiri) [reply src/styles/variables.scss:6]: She hasn't changed, being on the 8th lane

File: src/images/registrationWindow/close.svg
Comment (vlaskiri) [reply src/images/registrationWindow/close.svg:1]: Done

File: src/styles/variables.scss (hunk 5 above)
Comment (elzbthbrst) [reply src/styles/variables.scss:6]: u changed value from #aba8d4 to #ABA8D4, why?

Comment (vlaskiri) [reply src/styles/variables.scss:6]: We have such values ​​in Figma, and also in the entire file except for one variable. Out of context

File: src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (elzbthbrst) [reply src/components/registrationWindow/registrationAndEnter/loginOrRegister/ModalRegistration/ModalRegistration.jsx:8]: ok, if u think so

**Pull request merged**